src/utils/shared/redirects.js

```diff
@@ -52,7 +52,7 @@ const handleFrontRedirect = (host, pathname, clickEvent) => {
     }
 
     if (redirectUrl.startsWith('/')) {
-      redirectUrl = redirectUrl + location.search
+      redirectUrl = redirectUrl + location.search + location.hash
```

Comment:
why this change?

Reply:
It's necessary in dev mode when we have redirect which removes trailing slash. In this case we miss hash. But you are right - we shouldn't do it always. I added necessary condition.